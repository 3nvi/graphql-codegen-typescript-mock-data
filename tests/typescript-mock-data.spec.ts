import '@graphql-codegen/testing';

import { buildSchema, print } from 'graphql';
import { plugin } from '../src';

const testSchema = buildSchema(/* GraphQL */ `
    type Avatar {
        id: ID!
        url: String!
    }

    type User {
        id: ID!
        login: String!
        avatar: Avatar
    }

    type Query {
        user: User!
    }
`);

it('can be called', async () => {
    await plugin(testSchema, [], { typesFile: './types/graphql.ts' });
});

it('should generate mock data functions', async () => {
    const result = await plugin(testSchema, [], { typesFile: './types/graphql.ts' });

    expect(result).toBeDefined();
    // @ts-ignore
    expect(result).toMatchSnapshot();
});
