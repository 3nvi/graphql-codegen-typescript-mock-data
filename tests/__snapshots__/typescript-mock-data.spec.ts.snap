// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate mock data functions 1`] = `
"
export const anAvatar = (overrides?: Partial<Avatar>): Avatar => {
    return {
        id: '0550ff93-dd31-49b4-8c38-ff1cb68bdc38',
        url: 'aliquid',
        ...overrides
    };
};

export const aUser = (overrides?: Partial<User>): User => {
    return {
        id: 'a5756f00-41a6-422a-8a7d-d13ee6a63750',
        login: 'libero',
        avatar: anAvatar(),
        ...overrides
    };
};
"
`;

exports[`should generate mock data functions with external types file import 1`] = `
"/* eslint-disable @typescript-eslint/no-use-before-define,@typescript-eslint/no-unused-vars */
import { Avatar, User } from './types/graphql';

export const anAvatar = (overrides?: Partial<Avatar>): Avatar => {
    return {
        id: '0550ff93-dd31-49b4-8c38-ff1cb68bdc38',
        url: 'aliquid',
        ...overrides
    };
};

export const aUser = (overrides?: Partial<User>): User => {
    return {
        id: 'a5756f00-41a6-422a-8a7d-d13ee6a63750',
        login: 'libero',
        avatar: anAvatar(),
        ...overrides
    };
};
"
`;
